lib/socket.c

```diff
@@ -647,6 +687,13 @@ rpc_read_from_socket(struct rpc_context *rpc)
                                         return -1;
                                 }
 
+                                /*
+                                 * XXX adjust_inbuf() will allocate buffer equal to the
+                                 *     received PDU size though we will only use 1024 bytes.
+                                 *     This does not cause any correctness issue but the
+                                 *     extra allocation puts a burden on the allocator and
+                                 *     also the zeroing overhead. Fis this.
+                                 */
                                 if (adjust_inbuf(rpc, rpc->rm_xid[0]) != 0) {
                                         return -1;
                                 }
```

Comment:
We can limit the allocation to 1024 instead of rpc->rm_xid[0], since only that much we read into rpc->inbuf, rest goes in to user buffer as zero-copy.
@sahlberg ?

Reply:
I've done this change. @sahlberg , pls take a look